### DISCUSSION TITLE
window(cd) formatting

### DISCUSSION THREAD

**Discussion by ghost** [davatorium/rofi#2057] [Q&A]
Forgive me as I've been pouring over docs, examples online, sifting through other themes and configs trying to figure this out and my brain is mush at the time of this post...though it might be mush 24/7 anymore who can tell.

So I've been trying to find how to format window and windowcd results. All I can find is drun and combi stuff. I'd like to be able to do something like nice big icon on the left, multi line info on the right. Possibly with the Workspace bit last or hidden. It would be nice to hide the workspace if the application is on the workspace you are on and only display if the window is on a workspace other than the active one.

Right now window / windowcd wastes a lot of horizontal which would be nicer if you could stack the fields vertically (workspace, class, title/name) allowing for a nicer icon size too. So is it possible to do this? Can you hide workspace and class? Could you word wrap window titles/names rather than elipse them? Will the Mighty Duper Man get free from the clutches of Mr. Eval (get it? eval is bad...IS THIS THING ON?!)

**Comment by DaveDavenport** [comment]
What is unclear about the `window-format`? its listed in the manpage?

Wrapping is currently not possible.. elements need to have a fixed height. (for speed/simplicity reasons).

>  Will the Mighty Duper Man get free from the clutches of Mr. Eval (get it? eval is bad...IS THIS THING ON?!)

????

**Reply by ghost** [reply]
Nothing now that I've found it (local copy), I had searched the man page / wiki / other examples online and there was no window-format. As I said when I was searching the docs all I could find was a drun-format and combi-format.

So basically I can trim the fat but not so much with the stacked text formatting?

**Reply by DaveDavenport** [reply]
you can make it multi-row but its not 'great'..

```bash
rofi -show window -eh 2 -window-format '{t}
{n}'
```
will show title on first, name on second line in the element box that is now 2 lines high.

**Reply by ghost** [reply]
I was just fooling with the syntax. Thanks for the tip. I'm going to try to revise my theme to see if I can get things how I'd hoped. As for "not great" Hey it works so that's pretty great. I'd not have thought I could just do that. I had tried \n \r\n etc to no avail.

**Reply by DaveDavenport** [reply]
```bash
rofi -show window -eh 2 -window-format '{t}                
<b>{n}</b>'
```
if you want to do markup..  (https://docs.gtk.org/Pango/pango_markup.html#pango-markup)

it is easy to break stuff  this way :grinning:

**Reply by ghost** [reply]
Break it? I damn near bought it! wait...I... ;)
